src/daemon.rs

```diff
@@ -197,3 +199,34 @@ impl fmt::Display for NotifyState {
         }
     }
 }
+
+/// Perform some basic sanity checks against state entries.
+fn sanity_check_state_entries(state: &[NotifyState]) -> Result<(), SdError> {
+    for (index, entry) in state.iter().enumerate() {
+        match entry {
+            NotifyState::Fdname(ref name) => validate_fdname(name),
+            _ => Ok(()),
+        }
+        .map_err(|e| format!("invalid notify state entry #{}: {}", index, e))?;
+    }
+
+    Ok(())
+}
+
+/// Validate an `FDNAME` according to systemd rules.
+///
+/// The name may consist of arbitrary ASCII characters except control
+/// characters or ":". It may not be longer than 255 characters.
+fn validate_fdname(fdname: &str) -> Result<(), SdError> {
+    if fdname.len() > 255 {
+        return Err(format!("overlong fdname '{}'", fdname).into());
```

Comment:
Perhaps `fdname '{}' longer than 255 characters`?  This error message would be immediately actionable to a developer, whereas the current message requires at least one step through the documentation to find out what the maximum is.

Reply:
Good point, it read better that way.